cmd/soroban-cli/src/commands/tx/fetch/fee.rs

```diff
@@ -86,104 +97,152 @@ impl Cmd {
+    ) -> Result<(TransactionResult, TransactionMeta, TransactionEnvelope), Error> {
         let tx_result = resp.result.clone().ok_or(Error::None {
             field: "tx_result".to_string(),
-        })?; // fee charged
+        })?;
         let tx_meta = resp.result_meta.clone().ok_or(Error::None {
             field: "tx_meta".to_string(),
-        })?; // resource fees
+        })?;
         let tx_envelope = resp.envelope.clone().ok_or(Error::None {
             field: "tx_envelope".to_string(),
-        })?; // max fees
-
-        let fee_charged = tx_result.fee_charged;
-        let (non_refundable_resource_fee_charged, refundable_resource_fee_charged) =
-            Self::resource_fees_charged(&tx_meta);
-
-        let (max_fee, max_resource_fee) = Self::max_fees(&tx_envelope);
+        })?;
 
-        let resource_fee_charged =
-            non_refundable_resource_fee_charged + refundable_resource_fee_charged;
-        let inclusion_fee_charged = fee_charged - resource_fee_charged;
-        Ok(FeeTable {
-            fee_charged,
-            resource_fee_charged,
-            inclusion_fee_charged,
-            non_refundable_resource_fee_charged,
-            refundable_resource_fee_charged,
-            max_fee,
-            max_resource_fee,
-        })
+        Ok((tx_result, tx_meta, tx_envelope))
     }
 
-    fn max_fees(tx_envelope: &TransactionEnvelope) -> (i64, i64) {
+    fn extract_proposed_fees(tx_envelope: &TransactionEnvelope) -> ProposedFees {
         match tx_envelope {
             TransactionEnvelope::TxV0(transaction_v0_envelope) => {
-                let fee = transaction_v0_envelope.tx.fee;
-                (fee.into(), DEFAULT_FEE_VALUE)
+                let fee: i64 = transaction_v0_envelope.tx.fee.into();
+                ProposedFees {
+                    fee,
+                    resource_fee: DEFAULT_RESOURCE_FEE,
+                    inclusion_fee: fee - DEFAULT_RESOURCE_FEE,
+                }
             }
             TransactionEnvelope::Tx(transaction_v1_envelope) => {
                 let tx = transaction_v1_envelope.tx.clone();
-                let fee = tx.fee;
+                let fee: i64 = tx.fee.into();
                 let resource_fee = match tx.ext {
-                    xdr::TransactionExt::V0 => DEFAULT_FEE_VALUE,
+                    xdr::TransactionExt::V0 => DEFAULT_RESOURCE_FEE,
                     xdr::TransactionExt::V1(soroban_transaction_data) => {
                         soroban_transaction_data.resource_fee
                     }
                 };
-
-                (fee.into(), resource_fee)
+                ProposedFees {
+                    fee,
+                    resource_fee,
+                    inclusion_fee: fee - resource_fee,
+                }
             }
             TransactionEnvelope::TxFeeBump(fee_bump_transaction_envelope) => {
+                let inner_tx_resource_fee = match &fee_bump_transaction_envelope.tx.inner_tx {
```

Comment:
This is the main fix in this PR - instead of assuming all fee bump transactions have a resource fee of 0, like were we previously doing, we're now digging into the inner tx to see if its a soroban tx. If it is a classic tx, we can return the default resource fee, 0. If it's a soroban tx, we can return the inner tx's resource fee.